rdsbroker/broker_test.go

```diff
@@ -1703,4 +1709,125 @@ var _ = Describe("RDS Broker", func() {
+			BeforeEach(func() {
+				dbInstance.DescribeByTagDBInstanceDetails = []*awsrds.DBInstanceDetails{
+					&awsrds.DBInstanceDetails{
+						Identifier:     dbInstanceIdentifier,
+						Address:        "endpoint-address",
+						Port:           3306,
+						DBName:         "test-db",
+						MasterUsername: "master-username",
+						Engine:         "fake-engine",
+					},
+				}
+			})
+
+			It("should try to connect to databases", func() {
+				err := rdsBroker.CheckAndRotateCredentials()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(dbInstance.DescribeByTagCalled).To(BeTrue())
+				Expect(dbInstance.DescribeByTagKey).To(BeEquivalentTo("Broker Name"))
+				Expect(dbInstance.DescribeByTagValue).To(BeEquivalentTo(brokerName))
+				Expect(sqlProvider.GetSQLEngineCalled).To(BeTrue())
+				Expect(sqlProvider.GetSQLEngineEngine).To(BeEquivalentTo("fake-engine"))
+				Expect(sqlEngine.OpenCalled).To(BeTrue())
+				Expect(sqlEngine.OpenAddress).To(BeEquivalentTo("endpoint-address"))
+				Expect(sqlEngine.OpenPort).To(BeEquivalentTo(3306))
+				Expect(sqlEngine.OpenDBName).To(BeEquivalentTo("test-db"))
+				Expect(sqlEngine.OpenUsername).To(BeEquivalentTo("master-username"))
+			})
+
+			Context("and the passwords work", func() {
+				It("should not try to change the master password", func() {
+					err := rdsBroker.CheckAndRotateCredentials()
+					Expect(err).ToNot(HaveOccurred())
+					Expect(dbInstance.ModifyCalled).To(BeFalse())
+				})
+			})
+
+			Context("and the passwords don't work", func() {
+				BeforeEach(func() {
+					sqlEngine.PingError = errors.New("Login failed")
+				})
+
+				It("should try to change the master password", func() {
+					err := rdsBroker.CheckAndRotateCredentials()
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(dbInstance.ModifyCalled).To(BeTrue())
+					Expect(dbInstance.ModifyID).To(BeEquivalentTo(dbInstanceIdentifier))
+					Expect(dbInstance.ModifyDBInstanceDetails.MasterUserPassword).To(BeEquivalentTo(sqlEngine.OpenPassword))
+				})
+			})
+
+			Context("and there is an unkown ping error", func() {
+				BeforeEach(func() {
+					sqlEngine.PingError = errors.New("Unknown ping error")
+				})
+
+				It("should not try to change the master password", func() {
+					err := rdsBroker.CheckAndRotateCredentials()
+					Expect(err).To(HaveOccurred())
+					Expect(dbInstance.ModifyCalled).To(BeFalse())
+				})
+			})
+
+		})
+
+		Context("when we reset the password then try to bind", func() {
+			var (
+				bindDetails brokerapi.BindDetails
+			)
+
+			BeforeEach(func() {
+				dbInstance.DescribeByTagDBInstanceDetails = []*awsrds.DBInstanceDetails{
+					&awsrds.DBInstanceDetails{
+						Identifier:     dbInstanceIdentifier,
+						Address:        "endpoint-address",
+						Port:           3306,
+						DBName:         "test-db",
+						MasterUsername: "master-username",
+						Engine:         "fake-engine",
+					},
+				}
+
+				sqlEngine.PingError = errors.New("Login failed")
```

Comment:
It's not clear to me from the error checking in the test, or the descriptions of the `Context` and `It`, why this error is here. Could you explain?